### PR TITLE
More detailed LDAP docs

### PR DESCRIPTION
**Pull request by tristantarrant**

### CODE REVIEW

**Comment by dvagnero** [comment]
I'm working on docs. I'll add my commit tomorrow

**Comment by alvarolop** [comment]
Hello, thanks for this PR. The information expanded here regarding LDAP is great.
I don't see any more information to expand regarding LDAP.
I'll wait for the PR regarding the mappers!
Thank you for this!

**Comment by tristantarrant** [comment]
I've added a second commit with an explanation about role/permission mapping

File: documentation/src/main/asciidoc/topics/proc_configuring_cluster_role_mapper.adoc [hunk 1]
```diff
@@ -1,27 +1,37 @@
 [id='configuring-cluster-role-mapper_{context}']
-= Configuring the cluster role mapper
+= Configuring the cluster role and permission mappers
 
-The cluster role mapper maintains a dynamic mapping between principals and roles, by storing it in the cluster registry.
-You can manipulate these mappings using either the CLI or the REST API.
-By default, the mapping is performed using a strict string equivalence between principal names and roles. It is possible to configure the cluster role mapper to apply transformation to the principal name before performing a lookup.
+The cluster role mapper maintains a dynamic mapping between principals and roles.
+The cluster permission mapper maintains a dynamic set of role definitions.
```
Comment (wfink) [inline documentation/src/main/asciidoc/topics/proc_configuring_cluster_role_mapper.adoc:5]: From what I tested the permission mapper will have the same dynamic mapping as role-mapper, right?

File: documentation/src/main/asciidoc/topics/ref_role_permission_mappers.adoc [hunk 2]
```diff
@@ -1,18 +1,54 @@
...
 
-{brandname} users are implemented with the `javax.security.auth.Subject` class and represent a grouping of security principals of type `java.security.Principal`.
+{brandname} users are implemented as a collection of principals. Principals represent either an individual user identity, such as a username,
+or a group to which the user belongs. Internally, these are implemented with the `javax.security.auth.Subject` class.
+
+In order to perform authorization, principals must be mapped to role names, which in turn are expanded into a set of permissions.
 
 {brandname} includes the `PrincipalRoleMapper` API to associate security principals to roles and the `RolePermissionMapper` API to associate roles with sets of permissions.
-{brandname} also provides the following role and permission mapper implementations:
+{brandname} provides the following role and permission mapper implementations:
```
Comment (wfink) [inline documentation/src/main/asciidoc/topics/ref_role_permission_mappers.adoc:10]: Isn't it better to say 'uses ** created by CLI/REST'

**Review by wfink — CHANGES_REQUESTED** [review]

File: documentation/src/main/asciidoc/topics/ref_role_permission_mappers.adoc [hunk 3]
```diff
@@ -1,18 +1,54 @@
...
 
 Cluster role mapper:: Stores principal to role mappings in the cluster registry.
 
 Cluster permission mapper:: Stores role to permission mappings in the cluster registry and allows you to dynamically modify user roles and permissions.
 
 Identity role mapper:: Uses the principal name as the role name. The type or format of the principal name depends on the source. For example in an LDAP directory the principal name could be a Distinguished Name (DN).
```
[the remark this answers is not in the document]
Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/ref_role_permission_mappers.adoc:16]: Principal is both a user and a group name.

File: documentation/src/main/asciidoc/topics/proc_configuring_cluster_role_mapper.adoc (hunk 1 above)
Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/proc_configuring_cluster_role_mapper.adoc:5]: It's not possible to manipulate roles unless a MutableRolePermissionMapper is available (such as the ClusterPermissionMapper)

File: documentation/src/main/asciidoc/topics/ref_role_permission_mappers.adoc (hunk 2 above)
Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/ref_role_permission_mappers.adoc:11]: These docs also applied to embedded, so no.

**Comment by dvagnero** [comment]
Great work @tristantarrant  I added my commits. Let me know what you think.

File: documentation/src/main/asciidoc/topics/ref_role_permission_mappers.adoc [hunk 4]
```diff
@@ -1,26 +1,27 @@
...
-Identity role mapper:: Uses the principal name as the role name. The type or format of the principal name depends on the source. For example in an LDAP directory the principal name could be a Distinguished Name (DN).
+Identity role mapper:: Uses the principal name as the role name. The type or format of the principal name depends on the source. For example, in an LDAP directory the principal name could be a Distinguished Name (DN).
 
 Common name role mapper:: Uses the Common Name (CN) as the role name. You can use this role mapper with an LDAP directory or with client certificates that contain Distinguished Names (DN); for example `cn=managers,ou=people,dc=example,dc=com` maps to the `managers` role.
 
 NOTE: By default, principal-to-role mapping is only applied to principals which represent groups.
 It is possible to configure {brandname} to also perform the mapping for user principals.
 
-== A worked example
+== Mapping LDAP users to roles and permissions in {brandname}
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/topics/ref_role_permission_mappers.adoc:24]: I am merely using DNs here, but I wanted the example to showcase the mapping process, whatever the source of the principals

File: documentation/src/main/asciidoc/topics/proc_configuring_cluster_role_mapper.adoc (hunk 1 above)
Comment (wfink) [reply documentation/src/main/asciidoc/topics/proc_configuring_cluster_role_mapper.adoc:5]: ok, thats new and maybe not merged.
But a user might change the configuration later and then the stored roles or role-mappings are ignored

File: documentation/src/main/asciidoc/topics/proc_cli_user_roles_permissions.adoc [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+Describe roles with the [command]`user roles describe` command.
+
+[source,options="nowrap",subs=attributes+]
+----
+user roles describe simple
+{
+  "name" : "simple",
+  "permissions" : [ "ALL_READ","ALL_WRITE" ]
+}
+----
```
[the remark this answers is not in the document]
Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/proc_cli_user_roles_permissions.adoc:34]: That depends on https://github.com/infinispan/infinispan/pull/11114

File: documentation/src/main/asciidoc/topics/proc_configuring_cluster_role_mapper.adoc (hunk 1 above)
Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/proc_configuring_cluster_role_mapper.adoc:5]: Not new: https://github.com/infinispan/infinispan/blob/main/server/rest/src/main/java/org/infinispan/rest/resources/SecurityResource.java#L120 and https://github.com/infinispan/infinispan/blob/main/server/rest/src/main/java/org/infinispan/rest/resources/SecurityResource.java#L69

Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/proc_configuring_cluster_role_mapper.adoc:5]: Attempting to create a role with a RolePermissionMapper that does not implement MutableRolePermissionMapper will result in a `409 CONFLICT`

File: documentation/src/main/asciidoc/topics/ref_role_permission_mappers.adoc (hunk 4 above)
Comment (dvagnero) [reply documentation/src/main/asciidoc/topics/ref_role_permission_mappers.adoc]: Sure. I removed "LDAP" from the headings

**Comment by tristantarrant** [comment]
Thanks @dvagnero , I've squished the commits

**Review by dvagnero — APPROVED** [review]

**Review by wfink — APPROVED** [review]

**Comment by tristantarrant** [comment]
The paragraph

> NOTE: By default, principal-to-role mapping is only applied to principals which represent groups.
It is possible to configure {brandname} to also perform the mapping for user principals.

only applies if we backport https://github.com/infinispan/infinispan/pull/11120

**Pull request merged**